src/main/java/Storage.java

```diff
@@ -0,0 +1,96 @@
+import tasks.Deadline;
+import tasks.Event;
+import tasks.Task;
+import tasks.Todo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+import java.util.ArrayList;
+
+import java.util.Scanner;
+
+public class Storage {
+
+    private String filePath;
+
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Loads contents from the file and stores data in a list.
+     *
+     * @return list containing tasks from the file.
+     * @throws BlitzException if there's an error while creating the file
+     *     in case it doesn't exist.
+     */
+    public ArrayList<Task> loadFileContents() throws BlitzException {
+        ArrayList<Task> tasks = new ArrayList<Task>();
+        File f = new File(filePath);
+        try {
+            Scanner s = new Scanner(f);
+            while (s.hasNext()) {
+                String task = s.nextLine();
+                String[] keywords = task.split(" \\| ");
+                String done = keywords[1];
+                Task current = new Task("");
+                switch(task.charAt(0)) {
+                case 'T' :
+                    current = new Todo(keywords[2]);
+                    break;
+                case 'D' :
+                    current = new Deadline(keywords[2], LocalDateTime.parse(keywords[3],
+                            DateTimeFormatter.ofPattern("MMM d yyyy hh:mm a")));
+                    break;
+                default :
```

Comment:
Try to have the default statement be used for the intended default option, and not just to execute the last option.
> If there is no default action, you can use the default branch to detect errors.